src/lang/wasm.test.ts

```diff
@@ -1,13 +1,18 @@
 import { err } from 'lib/trap'
-import { parse } from './wasm'
+import { parse, ParseResult } from './wasm'
 import { enginelessExecutor } from 'lib/testHelpers'
+import { Node } from 'wasm-lib/kcl/bindings/Node'
+import { Program } from '../wasm-lib/kcl/bindings/Program'
 
 it('can execute parsed AST', async () => {
   const code = `x = 1
 // A comment.`
-  const ast = parse(code)
-  expect(err(ast)).toEqual(false)
-  const execState = await enginelessExecutor(ast)
-  expect(err(ast)).toEqual(false)
+  const result = parse(code)
+  expect(err(result)).toEqual(false)
+  const pResult = result as ParseResult
+  expect(pResult.errors.length).toEqual(0)
+  expect(pResult.program).not.toEqual(null)
+  const execState = await enginelessExecutor(pResult.program as Node<Program>)
```

Comment:
It's unfortunate that we need to do `as` here. I found a related issue, but it doesn't seem to have been resolved. https://github.com/vitest-dev/vitest/issues/2883 In general, I like to assert that it's not X (e.g. not `null`) and let it infer what it is. We could do that with type predicate helper functions or assertion functions, but I'm not sure it's worth the effort for a bunch of one-off things.

Reply:
Yeah, I think it's just in tests and it's a weakness of the `expect` stuff. I don't think there is a good solution, but I'm inclined not to care too much, since it's a test and the test will fail if the cast is incorrect